build warnings when HAVE_INET6 is off

Debian 11:

```
alfredh@debian:~/git/re$ make HAVE_INET6= HAVE_INET6_IGNORE_DEPRECATED=1 1>/dev/null
src/http/client.c: In function ‘http_client_set_laddr6’:
src/http/client.c:1111:46: warning: unused parameter ‘cli’ [-Wunused-parameter]
 1111 | void http_client_set_laddr6(struct http_cli *cli, const struct sa *addr)
      |                             ~~~~~~~~~~~~~~~~~^~~
src/http/client.c:1111:68: warning: unused parameter ‘addr’ [-Wunused-parameter]
 1111 | void http_client_set_laddr6(struct http_cli *cli, const struct sa *addr)
      |                                                   ~~~~~~~~~~~~~~~~~^~~~
src/dns/client.c: In function ‘dnsc_alloc’:
src/dns/client.c:859:19: warning: unused variable ‘laddr6’ [-Wunused-variable]
  859 |  struct sa laddr, laddr6;
      |                   ^~~~~~
src/sa/sa.c: In function ‘sa_set_scopeid’:
src/sa/sa.c:696:45: warning: unused parameter ‘scopeid’ [-Wunused-parameter]
  696 | void sa_set_scopeid(struct sa *sa, uint32_t scopeid)
      |                                    ~~~~~~~~~^~~~~~~
```

it would be nice if building with this option gave zero warnings ...

--- a/src/dns/client.c
+++ b/src/dns/client.c
@@ -856,7 +856,10 @@ int dnsc_alloc(struct dnsc **dcpp, const struct dnsc_conf *conf,
 	       const struct sa *srvv, uint32_t srvc)
 {
 	struct dnsc *dnsc;
-	struct sa laddr, laddr6;
+	struct sa laddr;
+#ifdef HAVE_INET6
+	struct sa laddr6;
+#endif
 	int err;
 
 	if (!dcpp)

--- a/src/http/client.c
+++ b/src/http/client.c
@@ -1113,5 +1113,8 @@ void http_client_set_laddr6(struct http_cli *cli, const struct sa *addr)
 #ifdef HAVE_INET6
 	if (cli && addr)
 		sa_cpy(&cli->laddr6, addr);
+#else
+	(void)cli;
+	(void)addr;
 #endif
 }

--- a/src/sa/sa.c
+++ b/src/sa/sa.c
@@ -703,6 +703,8 @@ void sa_set_scopeid(struct sa *sa, uint32_t scopeid)
 		return;
 
 	sa->u.in6.sin6_scope_id = scopeid;
+#else
+	(void)scopeid;
 #endif
 }
 

